server/Arcadia.Assistant.CSP/Vacations/ArcadiaVacationRegistry.cs

```diff
@@ -78,8 +87,35 @@ protected override void OnReceive(object message)
             }
         }
 
+        private async Task<Dictionary<string, double>> LoadEmployeeVacationDays()
+        {
+            var employeesTask = this.employeesQueryExecutor.Fetch();
+            var vacationsInfoTask = this.vacationsEmailLoader
+                .Ask<VacationsEmailLoader.GetVacationsInfo.Response>(VacationsEmailLoader.GetVacationsInfo.Instance);
+
+            await Task.WhenAll(employeesTask, vacationsInfoTask);
+
+            if (vacationsInfoTask.Result is VacationsEmailLoader.GetVacationsInfo.Error errorResult)
+            {
+                throw new Exception("Failed to load employee vacations info", errorResult.Exception);
+            }
+
+            var employees = employeesTask.Result;
+            var vacationsInfo = ((VacationsEmailLoader.GetVacationsInfo.Success)vacationsInfoTask.Result).EmployeeVacations;
```

Comment:
not really like the cast here. It won't throw compiler error in (i know) unlikely event of another added `GetVacationsInfo.Response` subtype. The safe (and readable) way would be to have a switch instead. However, result will be the same, an exception thrown and actor recreated, so i'll leave it to you :)

Reply:
Done.